executor/agents/agno/config_utils.py

```
@@ -9,17 +9,103 @@
+        return current[key]
+
+    # List index lookup
+    if isinstance(current, list) and key.isdigit() and int(key) < len(current):
+        return current[int(key)]
+
+    # Object attribute access (only for non-container objects)
+    if not isinstance(current, (dict, list)) and hasattr(current, key):
+        attr_value = getattr(current, key)
+        if callable(attr_value):
+            # Don't access callable methods (e.g., dict.items, list.append)
+            raise KeyError(f"Attribute {key} is callable")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)